fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -302,30 +306,58 @@ instance MonadLinkedIPFS Server where
   getLinkedPeers = asks ipfsRemotePeers
 
 instance MonadIPFSPinner Server where
-  pin cid = 
+  pin cid =
     asks clusterURL >>= \case
-      Nothing -> 
+      Nothing ->
         IPFS.Pin.add cid >>= \case
-          Left err -> 
-            return $ Error.openLeft err
-          Right _ -> 
-            return $ Right ()
+          Left  err -> return $ Error.openLeft err
+          Right _   -> return ok
 
       Just (IPFS.URL url) -> do
-        manager      <- asks httpManager
-        let 
-          query = Cluster.pinClient cid
+        manager <- asks httpManager
+
+        let
           env = mkClientEnv manager url
-        
-        (liftIO $ runClientM query env) >>= \case
+
+        liftIO (runClientM runPin env) >>= \case
+          Left err -> do
+            formattedErr <- Cluster.parseClientError err
+            return $ Error.openLeft formattedErr
+
+          Right _ ->
+            clusterCheckPin env
+
+    where
+      checkPin = Cluster.status cid
+      runPin   = Cluster.pin    cid
+      pause    = threadDelay 250_000 -- microseconds
```

Comment:
Let's just change to 500ms?

Reply:
Maybe use Dhority?

Reply:
Ooooooooh we need toactually rerun the whole request (Daniel said that it dropped around 10% of pins under high load)